contracts/OvertimeVaults/BaseSportVault.sol

```diff
@@ -0,0 +1,335 @@
+                        ONE;
+                    userInRound[round + 1][user] = true;
+                    usersPerRound[round + 1].push(user);
+                } else {
+                    balancesPerRound[round + 1][user] = 0;
+                    uint withdrawable = (balancesPerRound[round][user] * profitAndLossPerRound[round]) / ONE;
+                    sUSD.safeTransfer(user, withdrawable);
+                    withdrawalRequested[user] = false;
+                    userInRound[round + 1][user] = false;
+                    usersCurrentlyInVault = usersCurrentlyInVault - 1;
+                    emit Claimed(user, withdrawable);
+                }
+            }
+        }
+
+        // start next round
+        round += 1;
+
+        roundStartTime[round] = block.timestamp;
+
+        // allocation for next round doesn't include withdrawal queue share from previous round
+        allocationPerRound[round] = sUSD.balanceOf(address(this));
+        capPerRound[round + 1] = allocationPerRound[round];
+
+        emit RoundClosed(round - 1);
+    }
+
+    /// @notice Deposit funds from user into vault for the next round
+    /// @param amount Value to be deposited
+    function deposit(uint amount) external canDeposit(amount) {
+        sUSD.safeTransferFrom(msg.sender, address(this), amount);
+
+        uint nextRound = round + 1;
+
+        // new user enters the vault
+        if (balancesPerRound[round][msg.sender] == 0) {
+            require(usersCurrentlyInVault < maxAllowedUsers, "Max amount of users reached");
+            usersPerRound[nextRound].push(msg.sender);
+            userInRound[nextRound][msg.sender] = true;
+            usersCurrentlyInVault = usersCurrentlyInVault + 1;
+        }
+
+        balancesPerRound[nextRound][msg.sender] += amount;
+
+        // update deposit state of a user
+        depositReceipts[msg.sender] = DepositReceipt(nextRound, balancesPerRound[nextRound][msg.sender]);
+
+        allocationPerRound[nextRound] += amount;
+        capPerRound[nextRound] += amount;
+
+        emit Deposited(msg.sender, amount);
+    }
+
+    function withdrawalRequest() external {
+        require(vaultStarted, "Vault has not started");
+        require(!withdrawalRequested[msg.sender], "Withdrawal already requested");
+        require(balancesPerRound[round][msg.sender] > 0, "Nothing to withdraw");
+        require(balancesPerRound[round + 1][msg.sender] == 0, "Can't withdraw as you already deposited for next round");
+
+        uint nextRound = round + 1;
+        if (capPerRound[nextRound] > balancesPerRound[round][msg.sender]) {
+            capPerRound[nextRound] -= balancesPerRound[round][msg.sender];
+        }
+
+        withdrawalRequested[msg.sender] = true;
+        emit WithdrawalRequested(msg.sender);
+    }
+
+    /// @notice Set length of rounds
+    /// @param _roundLength Length of a round in miliseconds
+    function setRoundLength(uint _roundLength) external onlyOwner {
```

Comment:
maybe to add like owner method tests to increase coverage to max :) 

See coverage: https://app.codecov.io/gh/thales-markets/contracts/pull/219

Reply:
done